data-model/src/encoding/relativity.rs

```diff
@@ -0,0 +1,281 @@
+    }
+}
+
+// Entry <> Entry
+
+impl<const MCL: usize, const MCC: usize, const MPL: usize, N, S, PD>
+    RelativeEncoder<Entry<N, S, PathRc<MCL, MCC, MPL>, PD>>
+    for Entry<N, S, PathRc<MCL, MCC, MPL>, PD>
+where
+    N: NamespaceId + Encoder,
+    S: SubspaceId + Encoder,
+    PD: PayloadDigest + Encoder,
+{
+    /// Encode an [`Entry`] relative to a reference [`Entry`].
+    ///
+    /// [Definition](https://willowprotocol.org/specs/encodings/index.html#enc_etry_relative_entry).
+    async fn relative_encode<Consumer>(
+        &self,
+        reference: &Entry<N, S, PathRc<MCL, MCC, MPL>, PD>,
+        consumer: &mut Consumer,
+    ) -> Result<(), EncodingConsumerError<Consumer::Error>>
+    where
+        Consumer: BulkConsumer<Item = u8>,
+    {
+        let time_diff = self.timestamp.abs_diff(reference.timestamp);
+
+        let mut header: u8 = 0b0000_0000;
+
+        if self.namespace_id != reference.namespace_id {
+            header |= 0b1000_0000;
+        }
+
+        if self.subspace_id != reference.subspace_id {
+            header |= 0b0100_0000;
+        }
+
+        if self.timestamp > reference.timestamp {
+            header |= 0b0010_0000;
+        }
+
+        header |= CompactWidth::from_u64(time_diff).bitmask(4);
+
+        header |= CompactWidth::from_u64(self.payload_length).bitmask(6);
+
+        if let Err(err) = consumer.consume(header).await {
+            return Err(EncodingConsumerError {
+                bytes_consumed: 0,
+                reason: err,
+            });
+        };
+
+        if self.namespace_id != reference.namespace_id {
+            self.namespace_id.encode(consumer).await?;
+        }
+
+        if self.subspace_id != reference.subspace_id {
+            self.subspace_id.encode(consumer).await?;
+        }
+
+        self.path.relative_encode(&reference.path, consumer).await?;
+
+        encode_compact_width_be(time_diff, consumer).await?;
+
+        encode_compact_width_be(self.payload_length, consumer).await?;
+
+        self.payload_digest.encode(consumer).await?;
+
+        Ok(())
+    }
+}
+
+impl<const MCL: usize, const MCC: usize, const MPL: usize, N, S, PD>
+    RelativeDecoder<Entry<N, S, PathRc<MCL, MCC, MPL>, PD>>
+    for Entry<N, S, PathRc<MCL, MCC, MPL>, PD>
+where
+    N: NamespaceId + Decoder + std::fmt::Debug,
+    S: SubspaceId + Decoder + std::fmt::Debug,
+    PD: PayloadDigest + Decoder,
+{
+    /// Decode an [`Entry`] relative from this [`Entry`].
+    async fn relative_decode<Producer>(
+        &self,
+        producer: &mut Producer,
+    ) -> Result<Entry<N, S, PathRc<MCL, MCC, MPL>, PD>, DecodeError<Producer::Error>>
+    where
+        Producer: BulkProducer<Item = u8>,
+        Self: Sized,
+    {
+        let mut header_slice = [0u8];
+
+        producer
+            .bulk_overwrite_full_slice(&mut header_slice)
+            .await?;
+
+        let header = header_slice[0];
+
+        // Verify that bit 3 is 0 as specified - good indicator of invalid data.
```

Comment:
```suggestion
        // Verify that bit 3 is 0 as specified.
```

That second half made me open up the spec and wonder whether I missed anything.